src/function-builder-detectors/zisi.js

```diff
@@ -1,18 +1,142 @@
 const path = require('path')
 
-const { zipFunctions } = require('@netlify/zip-it-and-ship-it')
+const { zipFunction, zipFunctions } = require('@netlify/zip-it-and-ship-it')
 const del = require('del')
 const makeDir = require('make-dir')
+const pFilter = require('p-filter')
 
 const { getPathInProject } = require('../lib/settings')
 const { getFunctions } = require('../utils/get-functions')
 const { NETLIFYDEVERR } = require('../utils/logo')
 
-const bundleFunctions = async ({ config, sourceDirectory, targetDirectory }) => {
-  // @todo Build only the functions affected by the path that has changed.
-  await zipFunctions(sourceDirectory, targetDirectory, {
+const ZIP_CONCURRENCY = 5
+
+const zipFunctionsAndUpdateCache = async ({ cache, functions, sourceDirectory, targetDirectory, zipOptions }) => {
+  if (functions !== undefined) {
+    await pFilter(
```

Comment:
Using `p-filter` just for its capability of handling concurrency. We're effectively mimicking what `zipFunctions` does, but limiting it to a subset of functions. Hence the `ZIP_CONCURRENCY` constant using the same default value as `zip-it-and-ship-it`.